### ISSUE TITLE
[packager] Homebrew - Push to a different branch (or disable pushing)

### ISSUE THREAD

**Issue by gsmet** [jreleaser/jreleaser#1173]
Hi there,

I'm not sure if I'm making something bad here or if there is a bug (or if it's a missing feature).

For Quarkus, we are pushing maintenance branches from time to time. Meaning we have release 2.16.0.Final as the latest but we are still pushing 2.13 releases. It also happens that we release alpha versions of a coming branch.
In both cases, we do not want Homebrew `main` branch to be updated with this version as it should still be using the latest stable release.
It's not a problem for Chocolatey or SDKMAN as they are published as a different version.

I tried using:
```
export JRELEASER_BRANCH=${BRANCH}
if [ "$MAINTENANCE" == "true" ]; then
    export JRELEASER_HOMEBREW_GITHUB_BRANCH=${BRANCH}
fi
```

but it doesn't seem to change anything: the commit is still pushed to `main` and I have to revert it (see https://github.com/quarkusio/homebrew-tap/commits/main).

FWIW, I have the following warning when running JReleaser:
```
[WARN]  Auto configure detected '' as default branch but project has '3.0' configured
```

Not sure if I understood what `JRELEASER_HOMEBREW_GITHUB_BRANCH` is supposed to be doing.

Another way to solve my issue would be to have an env variable to disable a particular packager.

Thanks!

**Comment by aalmiray** [comment]
I can see the following validation in place

https://github.com/jreleaser/jreleaser/blob/5173229ad0a6e820f4ff9151d6c9c3d00f3cee79/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/validation/common/Validator.java#L235-L240

This should match the name you're using: `JRELEASER_HOMEBREW_GITHUB_BRANCH`. If it's not working as expected then this is indeed a bug. 

OTOH the warning message comes from 

https://github.com/jreleaser/jreleaser/blob/5173229ad0a6e820f4ff9151d6c9c3d00f3cee79/core/jreleaser-engine/src/main/java/org/jreleaser/engine/context/ModelConfigurer.java#L94-L97

Which makes me think `JRELEASER_BRANCH` has been set to an empty value (?). 

Finally, activating/deactivating packagers via env vars is something we've discusses internally before hand but have not found the use case for it. Now we have 😅 Matter of fact, other properties are also candidates for having their values resolved via env vars, also potentailly system props, the trick is defining the rules for resolution order.

**Comment by aalmiray** [comment]
@gsmet do you happen to have the capture of `trace.log` when the release was made? Alternatively you may run `jreleaser config` to check if branch value. I'll keep looking in my side.

**Comment by aalmiray** [comment]
Released in v1.5.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.5.0
